src/python/pants/backend/python/rules/python_test_runner.py

```diff
@@ -39,34 +39,6 @@
   all_targets = transitive_hydrated_targets.closure
```

Comment:
Changes to this file:
1) Removing the block `if not source_root_stripped_test_target_sources.snapshot.files` (early return)
2) Moving the resolve requirements logic back up to the top, where it originally was, so that this costly step gets cached earlier.